src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -563,14 +608,35 @@ export class NetworkAreaDiagramViewer {
             null,
             edgeMiddle,
             nodeRadius1,
-            nodeRadius2
+            nodeRadius2,
+            edgeType
         );
-        // redraw other voltage level node
-        const otherNode: SVGGraphicsElement | null =
-            edgeNodes[0]?.id == this.selectedElement?.id
-                ? edgeNodes[1]
-                : edgeNodes[0];
-        this.redrawOtherVoltageLevelNode(otherNode, [edge]);
+        // if dangling line edge -> redraw boundary node
+        let boundaryNodeMoved = false;
+        if (
+            edgeType == DiagramUtils.EdgeType.DANGLING_LINE &&
+            edgeNodes[1] != null
+        ) {
+            this.redrawBoundaryNode(
+                edgeNodes[1],
+                DiagramUtils.getAngle(edgeStart2, edgeMiddle),
+                nodeRadius2[1]
+            );
+            if (this.selectedElement?.id == edgeNodes[1].id) {
+                boundaryNodeMoved = true;
+            }
+        }
+        if (
+            edgeType != DiagramUtils.EdgeType.DANGLING_LINE ||
+            boundaryNodeMoved
```

Comment:
can you transform this into a else to have a clear separation danglingLine/not danglingLine? this probably means removing `edgeNodes[1] != null` in the previous if condition

Reply:
done